packages/node/src/dev-server.mts

```diff
@@ -6,18 +6,18 @@ if (!entrypoint) {
 }
 
 import { join } from 'path';
-const useRequire = process.env.VERCEL_DEV_IS_ESM !== '1';
-
 import type { Headers } from 'node-fetch';
-import type { VercelProxyResponse } from './types';
+import type { VercelProxyResponse } from './types.js';
 import { Config } from '@vercel/build-utils';
-import { createEdgeEventHandler } from './edge-functions/edge-handler';
+import { createEdgeEventHandler } from './edge-functions/edge-handler.mjs';
 import { createServer, IncomingMessage, ServerResponse } from 'http';
-import { createServerlessEventHandler } from './serverless-functions/serverless-handler';
-import { EdgeRuntimes, isEdgeRuntime, logError } from './utils';
+import { createServerlessEventHandler } from './serverless-functions/serverless-handler.mjs';
+import { EdgeRuntimes, isEdgeRuntime, logError } from './utils.js';
 import { getConfig } from '@vercel/static-config';
 import { Project } from 'ts-morph';
-import listen from 'async-listen';
+import asyncListen from 'async-listen';
+
+const { default: listen } = asyncListen;
```

Comment:
I showed this in the virtual coffee 4/21. Since this file is a `.mts` file (e.g. a real ESM module), it doesn't resolve static default export in CommonJS files. You can't destructure `import { default as listen } ...` either.

Since this authoring of that code, `async-listen` has been "fixed" to export `listen()` as a named export: https://github.com/vercel/async-listen/releases/tag/2.1.0. I could update the `async-listen` dependency and in theory the code would be destructurable like `import { listen } from 'async-listen'`, but not 100% certain.

Reply:
Ohh, right. This was that weird case.